src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Core/SymbolIsBannedAnalyzerBase.cs

```diff
@@ -329,6 +342,31 @@ void VerifyDocumentationSyntax(Action<Diagnostic> reportDiagnostic, SyntaxNode s
             }
         }
 
+        private static IEnumerable<INamespaceSymbol> GetContainingNamespaces(ISymbol symbol)
```

Comment:
This method seems to have only one call site. Consider making it a local function instead.

Reply:
@mavasani I would love an analyzer that detected this case and suggested it in the ide. Any thoughts on adding one to this repo?

Reply:
errrrrrr.  that's pushing a particular style *hard* (especially causing deep nesting of helpers).  I def would be wary of taht.  :)

Reply:
It wouldn't have to be on by default. I would consider adding it as a suggestion for dotnet/project-system.

Personally I have no problem with deep nesting. I'd rather have the scope in which a function is used constrained as tightly as possible so I don't have to reason about its use in broader contexts.

Reply:
Done.